Mobile/app/src/main/java/com/lesenia/mobile/LoginActivity.java

```diff
@@ -0,0 +1,115 @@
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+        fieldsInit();
+
+        loginButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                final String email = Objects.requireNonNull(Objects.requireNonNull(emailField.getText()).toString().trim());
+                final String password = Objects.requireNonNull(passwordField.getText()).toString().trim();
+                logIn(email, password);
+            }
+        });
+        signUpLink.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                startActivity(new Intent(LoginActivity.this, RegisterActivity.class));
+            }
+        });
+    }
+
+    private void fieldsInit() {
+        auth = FirebaseAuth.getInstance();
+        emailField = findViewById(R.id.login_email);
+        passwordField = findViewById(R.id.login_password);
+        emailFieldLayout = findViewById(R.id.login_layout_email);
+        passwordFieldLayout = findViewById(R.id.login_layout_password);
+        loginButton = findViewById(R.id.login_loginButton);
+        signUpLink = findViewById(R.id.login_signUpLink);
+    }
+
+    private void logIn(final String email, final String password) {
+
+        if (isDataValid(email, password)) {
+            Task<AuthResult> authResultTask = auth.signInWithEmailAndPassword(email, password)
+                    .addOnCompleteListener(LoginActivity.this, new OnCompleteListener<AuthResult>() {
+                        @Override
+                        public void onComplete(@NonNull Task<AuthResult> task) {
+                            if (task.isSuccessful()) {
+                                startActivity(new Intent(LoginActivity.this, UserActivity.class));
+                            } else {
+                                loginError();
+                            }
+                        }
+                    });
+        }
+    }
+
+    private boolean isEmailValid(final String email) {
+        if (email.isEmpty()) {
+            emailFieldLayout.setError(getString(R.string.enter_email));
+            return false;
+        } else {
+            emailFieldLayout.setError(null);
+            return true;
+        }
+    }
+
+    private boolean isPasswordValid(final String password) {
+        if (password.isEmpty() || password.length() < 8) {
+            passwordFieldLayout.setError(getString(R.string.invalid_password));
+            return false;
+        } else {
+            passwordFieldLayout.setError(null);
+            return true;
+        }
+    }
+
+    private boolean isDataValid(final String email, final String password) {
+        boolean emailValid = isEmailValid(email);
+        boolean passwordValid = isPasswordValid(password);
+
+        return emailValid && passwordValid;
```

Comment:
You can simplify: return isEmailValid(email) && isPasswordValid(password).

Reply:
Fixed , thank you!